src/app/hooks/useTxErrorMessage.ts

```diff
@@ -0,0 +1,18 @@
+import { useTranslation } from 'react-i18next'
+import { TxError } from '../../oasis-nexus/api'
+
+export const useTxErrorMessage = (error: TxError | undefined): string | undefined => {
+  const { t } = useTranslation()
+  if (!error) return undefined
+  if (error.code === 8) {
```

Comment:
This is a nice quick fix, but I think we really should discuss with BE, if they want to add support for this on Nexus.

Reply:
What does code 8 mean? Any docs?

Reply:
> All the errors are defined per module in the oasis sdk repository 

this was posted by @kostko in Slack

Reply:
> Evm code 8 is reverted :) 

> there is a large amount of error codes and modules. But one thing about evm/8 is that the contract can customize the revert error message. And it is also possible for it to be empty (which is the one that nexus/explorer don’t handle in the best way currently)

> All the errors are defined per module in the oasis sdk repository

> But most others don’t have custom error messages so you will already get the right thing :)

> So the error message for the above returned by the runtime is actually “reverted: ” because thr custom part comes after the colon, but nexus is using a custom decoder for that since when the custom message exists, it is evm abi and base64 encoded. So it is stripping the “reverted: “ part

Reply:
Oh this is actually wrong, it shouldn't just match on code 8, it also needs to match the module. Error codes are namespaced per module, e.g. evm/8 is defined [here](https://github.com/oasisprotocol/oasis-sdk/blob/9fb148a52636d7c1e8f839d1caa97aceef449e04/runtime-sdk/modules/evm/src/lib.rs#L157-L159).

Reply:
It has been discussed with @donouwens and @kostko that adding back the "Reverted" part, and saying "without a message" is the right course of action for now.

Reply:
>  it shouldn't just match on code 8, it also needs to match the module. 

Updated accordingly.